src/ray/core_worker/core_worker_process.cc

```diff
@@ -79,21 +112,57 @@ CoreWorkerProcessImpl::CoreWorkerProcessImpl(const CoreWorkerOptions &options)
                      ? ComputeDriverIdFromJob(options_.job_id)
                      : WorkerID::FromRandom()) {
   if (options_.enable_logging) {
-    std::stringstream app_name_ss;
-    app_name_ss << LanguageString(options_.language) << "-core-"
-                << WorkerTypeString(options_.worker_type);
-    if (!worker_id_.IsNil()) {
-      app_name_ss << "-" << worker_id_;
+    // Setup logging for worker system logging.
+    {
+      std::stringstream app_name_ss;
+      app_name_ss << LanguageString(options_.language) << "-core-"
+                  << WorkerTypeString(options_.worker_type);
+      if (!worker_id_.IsNil()) {
+        app_name_ss << "-" << worker_id_;
+      }
+      const std::string app_name = app_name_ss.str();
+      const std::string log_filepath =
+          RayLog::GetLogFilepathFromDirectory(options_.log_dir, /*app_name=*/app_name);
+      RayLog::StartRayLog(app_name,
+                          RayLogLevel::INFO,
+                          log_filepath,
+                          /*err_log_filepath=*/"",
+                          ray::RayLog::GetRayLogRotationMaxBytesOrDefault(),
+                          ray::RayLog::GetRayLogRotationBackupCountOrDefault());
+    }
+
+    // Setup logging for worker application logging.
+    if (options_.worker_type == WorkerType::WORKER && !IsEnvSet("RAY_LOG_TO_STDERR")) {
```

Comment:
We need this for SPILL_WORKER and RESTORE_WORKER as well basically anything that starts with default_worker.py